src/components/TodoItem.jsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
+import styles from '../styles/TodoItem.module.css';
+
+const TodoItem = ({
+  todo, handleChange, handleDelete, setUpdate,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const viewMode = {};
+  const editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdateDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  return (
+    <>
+      <li className={styles.item}>
+        <div className={styles.content} style={viewMode}>
+          <input
+            type="checkbox"
+            checked={todo.completed}
+            onChange={() => handleChange(todo.id)}
+          />
+          <button type="submit" onClick={handleEditing}>Edit</button>
+          <button type="submit" onClick={() => handleDelete(todo.id)}>Delete</button>
```

Comment:
You may add SVG icon for Delete instead of using `Delete` text.
Your Design:
![image](https://github.com/manzitresor/react_todo_app/assets/24216198/069e0fd1-3ccf-401f-9f58-1947d059e9ed)
Tutorial Design:
![image](https://github.com/manzitresor/react_todo_app/assets/24216198/0917ff18-341b-4cc2-a8a2-805baf9b1ec4)